packages/scm/src/common/scm.ts

```diff
@@ -0,0 +1,126 @@
+/********************************************************************************
+ * Copyright (C) 2019 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import URI from '@theia/core/lib/common/uri';
+import { Disposable, Event } from '@theia/core/lib/common';
+
+export interface ScmResourceDecorations {
+    icon?: URI;
```

Comment:
If this will be used by the backend too, then you have to use `string` instead of the `URI`, if no, you can put it under `browser`.

Reply:
moved to `browser`